tests/conftest.py

```diff
@@ -452,13 +452,16 @@ def make_config_file(tmpdir_cwd):
     """
 
     def _maker(
-        pyproject_param: str, new_default: Any, config_file_name: str = CONFIG_FILE_NAME
+        pyproject_param: str,
+        new_default: Any,
+        config_file_name: str = DEFAULT_CONFIG_FILE_NAMES[0],
     ) -> Path:
-        # Make a config file with this one config default override
-        config_path = tmpdir_cwd / pyproject_param
-        config_file = config_path / config_file_name
-        config_path.mkdir(exist_ok=True)
+        # Create a nested directory structure if config_file_name includes directories
+        config_dir = Path(tmpdir_cwd / config_file_name).parent
```

Comment:
```suggestion
        config_dir = (tmpdir_cwd / config_file_name).parent
```

The `tmpdir_cwd` already yields a Path-like object, so it's unnecessary.

Reply:
Ah, no. Can', because `as_posix()` shows full path and breaks tests.